### DISCUSSION TITLE
Releasing the next version

### DISCUSSION THREAD

**Discussion by ahockersten** [openliero/openliero#12] [General]
Hey. So, I started drawing up some very rough plans for what I want to do with openliero. Specifically I want to release something before I lose steam (which will undoubtedly happen when I start my new job on Sep 2, if it doesn't happen before that).

What would I want before releasing something? I think I would be fine with releasing what we have today, but I have a slightly grandioser plan of things I want in place before releasing:
- some maps from the community (10-20 or so)
- maybe some additional "match setup" settings (if someone has any)
- a bunch of "interesting to work on" issues on the issue tracker prepared (I have a few in mind)

My hope by adding the above is that we can get some more people interested in hacking on Liero. The risk is of course getting patches that are more work than they are worth, but that's a risk I'm willing to take.

So my idea is to prepare those "interesting to work on" issues, and after that ask for maps/settings/testing of the latest builds on the Liero Discord. If there are no big issues, we release latest master as 2.0.0. This is because I like big numbers, and there are unofficial 1.37-1.41(?) versions out there, like the Orbmit versions, that could potentially cause confusion.

As for what comes after openliero 2.0.0, I have no idea. It depends a lot on the response on the above I suppose.

Thoughts? 🙂

**Comment by grawlinson** [comment]
Chucking in my wishlist:

* Allow the game to be installed globally (via Linux package managers, or different avenues for other platforms), this will require refactoring of resource loading & a re-think of how assets are handled. e.g. global (read-only) path for TCs; have openliero traverse said path for a list of TCs, displayed, followed by a list of TCs found in a user path. Related, I'd like all extra TCs/config/settings saved in per-platform user directories so that they can be edited without worrying about breaking the vanilla game.
* Reduce the amount of ~~bullshit~~ friction from the current build process.
* Somewhat linked to above; make the code a hell of a lot nicer to read (more comments/documentation is A++) as well as reducing the amount of platform-specific hackery in favour of offloading said hackery to standard libraries.
* I'd like to chuck external dependencies in a `libs/` directory found in the root so that we can statically link against these for platforms where distributing shared libraries is a  pain (looking at you, Windows & macOS). I would also like to add compile-time flags for system-provided libraries _if_ consumers do not want to statically link said libraries.
* I abhor dealing with code/resource licensing/copyright issues because they are a _huge_ blocker, I mean, just look at the issues we currently have with distributing the original WAVs. Before pulling in any external code/resources, it needs to be explicitly clear _what_ we are going to be doing with these assets. One of my moonshot wishes would be to release openliero on Steam for free.
* More :unicorn:.

Realistically, none of these are getting done before @ahockersten starts his new job, so let's temper our expectations!

**Comment by ahockersten** [comment]
For Windows, I think packaging the SDL DLLs like we do today makes sense, we don't need to be able to statically link there (unless there are more dependencies we need to add). Otherwise I agree on everything you've said.

I would also like to release on Steam at some (realistically probably far) point into the future. One prerequisite of that we should solve ASAP is adding some kind of license. This unfortunately involves getting previous contributors to sign off on that new license.

Now, however, we should decide on a license for future contributions so we at least get that settled before there's even more work on resolving licensing issues. 🤮

I don't have a strong preference for license. I think at least some things (like the lierolibre sounds) are under a WTFPL license, but I might want something a bit more serious with attribution, like MIT or BSD. But I wouldn't object to going for GPL either.

**Reply by grawlinson** [reply]
> I don't have a strong preference for license. I think at least some things (like the lierolibre sounds) are under a WTFPL license, but I might want something a bit more serious with attribution, like MIT or BSD. But I wouldn't object to going for GPL either.

liero.be has the licenses for various versions of Liero at the footer, here's the relevant ones:

* [Liero 1.33](https://liero.be/licenses/133/license.txt)
* [Liero 1.35](https://liero.be/licenses/135/license.txt)
* [lierolibre](https://liero.be/licenses/libre/license.txt)

From my understanding, openliero is essentially BSD-2-Clause as lierolibre was initially forked from gliptic/liero anyway.

**Reply by grawlinson** [reply]
I've added the BSD-2-Clause license in #18.

**Comment by ahockersten** [comment]
> (...)  refactoring of resource loading & a re-think of how assets are handled.
I made an issue for that here: https://github.com/openliero/openliero/issues/17
